src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -246,7 +247,21 @@
 
   }
 
-  public HdfsSinkConnectorConfig(Map<String, String> props) {
+  public static HdfsSinkConnectorConfig getConfig(Map<String, String> props) {
+    return getConfig(CONFIG_DEF, props);
+  }
+
+  protected static HdfsSinkConnectorConfig getConfig(
+      ConfigDef configDef,
+      Map<String, String> props
+  ) {
+    ConcurrentMap<String, String> propsCopy = new ConcurrentHashMap<>(props);
+    propsCopy.putIfAbsent(StorageCommonConfig.STORAGE_CLASS_CONFIG, HdfsStorage.class.getName());
+    propsCopy.putIfAbsent(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG, AvroFormat.class.getName());
+    return new HdfsSinkConnectorConfig(configDef, propsCopy);
+  }
+
+  protected HdfsSinkConnectorConfig(Map<String, String> props) {
```

Comment:
What about keeping the public constructor and instead adding the default properties if missing:

```
public HdfsSinkConnectorConfig(Map<String, String> props) {
    this(CONFIG_DEF, addDefaults(props));
}

private static Map<String, String> addDefaults(Map<String, String> props) {
    ConcurrentMap<String, String> propsCopy = new ConcurrentHashMap<>(props);
    propsCopy.putIfAbsent(StorageCommonConfig.STORAGE_CLASS_CONFIG, HdfsStorage.class.getName());
    propsCopy.putIfAbsent(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG, AvroFormat.class.getName());
    return propsCopy;
}
```

The benefit is that the public constructor stays the same, just in case people extend this connector/class. Not sure if that's a worry. But it would reduce the number of changes where the constructor is normally used.